components/ActiveLocalizedLink.tsx

```diff
@@ -0,0 +1,60 @@
+import { useRouter } from 'next/router';
+import { useState, useEffect } from 'react';
+import LocalizedLink from './LocalizedLink';
+import type { LinkProps } from 'next/link';
+import type { PropsWithChildren } from 'react';
+
+type ActiveLocalizedLinkProps = LinkProps & {
+  className?: string;
+  activeClassName: string;
+};
+
+const ActiveLocalizedLink = ({
+  children,
+  activeClassName,
+  className,
+  ...props
+}: PropsWithChildren<ActiveLocalizedLinkProps>) => {
+  const { asPath, isReady } = useRouter();
+  const [computedClassName, setComputedClassName] = useState(className);
+
+  useEffect(() => {
```

Comment:
I wonder how many times this effect runs for each link? Could you give some benchmarks if possible (simple console.logs) should be enough (from the initialisation of the Storybook -> Last re-render)

Reply:
Not sure what you mean by **Storybook -> Last re-render**, but I added a `console.log` statement and it ran one time for each link.

Reply:
Nice